pkg/kubelet/deviceplugin/endpoint.go

```diff
@@ -58,7 +59,8 @@ func newEndpoint(socketPath, resourceName string, callback MonitorCallback) (*en
 	ctx, stop := context.WithCancel(context.Background())
 
 	return &endpoint{
-		client: client,
+		client:     client,
+		clientConn: c,
```

Comment:
I am trying to understand what extra resources will be released through grpc client Stop given that we already call context cancel that should close network connection AFAIU. Looking at the grpc document: 
https://github.com/grpc/grpc-go/blob/master/clientconn.go#L317
Looks like we should just call client Stop() after Dial succeeds. So probably we only need to keep grpc clientConn in endpoint struct and skip context and context callback?

Reply:
> given that we already call context cancel that should close network connection AFAIU

This closes the stream not the connexion. 
Which means that pretty quickly device plugins won't be able to connect to kubelet if one device plugin continuously reconnects.

Reply:
Looking at grpc.ClientConn.Close(), I think it closes all network connections associated with the grpc client:
https://github.com/grpc/grpc-go/blob/master/clientconn.go#L820
So I wonder whether we still need to keep connection cancel function.